### DISCUSSION TITLE
SetParent

### DISCUSSION THREAD

**Discussion by keinou** [halildurmus/win32#195] [Q&A]
Example for SetParent usage?

For example, run notepad and SetParent this inside a Container()

**Comment by timsneath** [comment]
Can you explain further? Are you asking about how to use the Win32 `SetParent` API in general? 

Or are you asking about hosting a window inside a Flutter desktop app?

**Comment by keinou** [comment]
In C# to put an external program inside a panel we use SetParent, what I need to do is the same thing, for example:

Run any program, like paint or notepad and after that take your window into my application, so that it is "docked", for example:

![image](https://user-images.githubusercontent.com/20158261/113032902-67d68f00-915e-11eb-95a0-1c9c366b788d.png)

**Reply by timsneath** [reply]
So you can call the `SetParent` method from Dart just like you can in C#. If you've created two Win32 window objects, you can reparent them using this command. 

But if what you're talking about is making it a child window in *Flutter*, that's not possible, because Flutter widgets aren't child windows in the same way that traditional Win32 or WinForms windows are. In the image you've included, I assume you're using something like a Panel object that you can get an HWND for, and then using that as the parent for the Paint process. 

This isn't a good idea, even here, incidentally, as Raymond Chen's article describes:
https://devblogs.microsoft.com/oldnewthing/20130412-00/?p=4683

TL;DR The `SetParent` command is really intended for moving windows within a same process. 

There is no concept of a window handle in Flutter for anything other than the parent window. The contents of the window are essentially one giant canvas to which Flutter paints all its controls. I think this issue https://github.com/flutter/flutter/issues/31713 addresses the use case presented above of a platform view that can paint native Windows content, but it's not implemented in any current Flutter release.
